src/interface.jl

```diff
@@ -0,0 +1,655 @@
+    f2 = forward(b2, f1.rv)
+    return (rv=f2.rv, logabsdetjac=f2.logabsdetjac .+ f1.logabsdetjac .+ f.logabsdetjac)
+end
+function _forward(f::NamedTuple, b::Bijector, bs::Bijector...)
+    f1 = forward(b, f.rv)
+    f_ = (rv=f1.rv, logabsdetjac=f1.logabsdetjac .+ f.logabsdetjac)
+    return _forward(f_, bs...)
+end
+_forward(x, b::Bijector, bs::Bijector...) = _forward(forward(b, x), bs...)
+forward(cb::Composed{<: Tuple}, x) = _forward(x, cb.ts...)
+
+function forward(cb::Composed, x)
+    rv, logjac = forward(cb.ts[1], x)
+    
+    for t in cb.ts[2:end]
+        res = forward(t, rv)
+        rv = res.rv
+        logjac = res.logabsdetjac .+ logjac
+    end
+    return (rv=rv, logabsdetjac=logjac)
+end
+
+##############################
+# Example bijector: Identity #
+##############################
+
+struct Identity <: Bijector end
+(::Identity)(x) = x
+(::Inversed{Identity})(y) = y
+
+forward(::Identity, x) = (rv=x, logabsdetjac=zero(x))
+
+logabsdetjac(::Identity, y::T) where T <: Real = zero(T)
+logabsdetjac(::Identity, y::AbstractArray{T}) where T <: Real = zero(T)
+
+const IdentityBijector = Identity()
+
+###############################
+# Example: Logit and Logistic #
+###############################
+using StatsFuns: logit, logistic
+
+struct Logit{T<:Real} <: Bijector
+    a::T
+    b::T
+end
+
+(b::Logit)(x) = @. logit((x - b.a) / (b.b - b.a))
+(ib::Inversed{<: Logit{<: Real}})(y) = @. (ib.orig.b - ib.orig.a) * logistic(y) + ib.orig.a
+
+logabsdetjac(b::Logit{<:Real}, x) = @. log((x - b.a) * (b.b - x) / (b.b - b.a))
+
+#############
+# Exp & Log #
+#############
+
+struct Exp <: Bijector end
+struct Log <: Bijector end
+const exp_b = Exp()
+const log_b = Log()
+
+(b::Log)(x) = @. log(x)
+(b::Exp)(y) = @. exp(y)
+
+inv(b::Log) = exp_b
+inv(b::Exp) = log_b
+
+logabsdetjac(b::Log, x) = log(x)
+logabsdetjac(b::Exp, y) = - y
+
+#################
+# Shift & Scale #
+#################
+struct Shift{T} <: Bijector
+    a::T
+end
+
+(b::Shift)(x) = b.a + x
+(b::Shift{<: Real})(x::AbstractArray) = b.a .+ x
+(b::Shift{<: AbstractVector})(x::AbstractMatrix) = b.a .+ x
+
+inv(b::Shift) = Shift(-b.a)
+logabsdetjac(b::Shift, x) = zero(eltype(x))
+# FIXME: ambiguous whether or not this is actually a batch or whatever
+logabsdetjac(b::Shift{<: Real}, x::AbstractMatrix) = zeros(eltype(x), size(x, 2))
+logabsdetjac(b::Shift{<: AbstractVector}, x::AbstractMatrix) = zeros(eltype(x), size(x, 2))
+
+struct Scale{T} <: Bijector
+    a::T
+end
+
+(b::Scale)(x) = b.a * x
+(b::Scale{<: Real})(x::AbstractArray) = b.a .* x
+(b::Scale{<: AbstractVector{<: Real}})(x::AbstractMatrix{<: Real}) = x * b.a
+
+inv(b::Scale) = Scale(inv(b.a))
+inv(b::Scale{<: AbstractVector}) = Scale(inv.(b.a))
+
+# TODO: should this be implemented for batch-computation?
+# There's an ambiguity issue
+#      logabsdetjac(b::Scale{<: AbstractVector}, x::AbstractMatrix)
+# Is this a batch or is it simply a matrix we want to scale differently
+# in each component?
```

Comment:
Does the behaviour different for this two cases?

Reply:
In one case we want to return a vector of values with length as long as the batch-size, while in the other case we simply want to return a single number, right?

Reply:
A good example is the `logabsdetjac` for `log` and `exp` transforms. It makes sense to let these act on vectors elementwise, etc. but when computing the `logabsdetjac(::Log, x::Vector)` we need to sum over the entires; but what if `x` is actually a vector representing a _batch_, then we should _not_ sum.

Reply:
What we could do is also parametrize `Scale` with an additional type, specifying the expected _input/output_ and then we can dispatch on this. E.g. for the `Exp` and `Log` we can do

```julia
struct Exp{T} <: Bijector end
struct Log{T} <: Bijector end
const exp_b = Exp{Real}()
const log_b = Log{Real}()

(b::Log)(x) = @. log(x)
(b::Exp)(y) = @. exp(y)

inv(b::Log{T}) where {T} = Exp{T}()
inv(b::Exp{T}) where {T} = Log{T}()

logabsdetjac(b::Log, x) = sum(log.(x))
logabsdetjac(b::Exp, y) = - sum(y)

logabsdetjac(b::Log{<: Real}, x::AbstractVector) = log.(x)
logabsdetjac(b::Exp{<: Real}, y::AbstractVector) = - y
logabsdetjac(b::Log{<: AbstractVector}, x::AbstractMatrix) = sum(log.(x); dims = 1)
logabsdetjac(b::Exp{<: AbstractVector}, y::AbstractMatrix) = - sum(y; dims = 1)
```

This leads to nice method-implementations using multiple dispatch, but feels very hacky :thinking:

Reply:
OR we could adapt something similar to `Multivariate, Univariate`, etc. in `Distributions`. Then the above would be 
```julia
struct Scale{T, V} <: Bijector
    a::T
end

# sane defaults
(b::Scale)(x) = b.a * x
(b::Scale{<:Real})(x::AbstractArray) = b.a .* x

logabsdetjac(b::Scale, x) = log(abs(b.a))

# more specialized
function logabsdetjac(b::Scale{<:Real, Multivariate}, x::AbstractMatrix)
    return log(abs(b.a)) .* ones(size(x, 2))
end

Reply:
> In one case we want to return a vector of values with length as long as the batch-size, while in the other case we simply want to return a single number, right?

Yes I get your point. Thanks.

The parametric idea looks promising. We prefer the one with `AbstractArray` as it also allows us to put `CuArray` which indicates the device.

Reply:
> The parametric idea looks promising. We prefer the one with AbstractArray as it also allows us to put CuArray which indicates the device.

That's an interesting idea; didn't think of that :+1: